### ISSUE TITLE
Closed route-data spec validation doesn't allow :conflicting key

### ISSUE THREAD

**Issue by Deraen** [metosin/reitit#376]
Closed route-data spec validation doesn't allow :conflicting key

https://cljdoc.org/d/metosin/reitit/0.4.2/doc/basics/route-conflicts#route-conflicts

**Comment by wandersoncferreira** [comment]
Hello @Deraen could you give me more information about this bug? I would like to investigate further.

**Comment by ikitommi** [comment]
the added `:conflicting` key is not defined in `reitit.spec`.

**Comment by dawran6** [comment]
@wandersoncferreira are you still working on this one? I'm happy to take if over if not :).
Based on the description, I created a minimal example to demonstrate the issue:

```
(r/router
  ["a" {:conflicting true}]
  {:validate rs/validate
   :spec     ::rs/default-data
   ::rs/wrap spec-tools.spell/closed})

=>
Invalid route data:

-- On route -----------------------

"a"

{:conflicting true}

:conflicting - failed: #{:no-doc :name :handler} in: [:conflicting 0] at: [0]
	 :expound.spec.problem/type  :spec-tools.spell-spec.alpha/unknown-key
	 :spec-tools.spell-spec.alpha/unknown-key  :conflicting

```

For anyone wants to get rid of this error before the fix is released, the [documentation on Closed Specs](https://cljdoc.org/d/metosin/reitit/0.5.6/doc/basics/route-data-validation#closed-specs) hinted the solution to this problem is to add additional spec as you needed with `s/merge`:

```
(s/def ::conflicting boolean?)
(r/router
  ["a" {:conflicting true}]
  {:validate rs/validate
   :spec     (s/merge ::rs/default-data
                      (s/keys :req-un [::conflicting]))
   ::rs/wrap spec-tools.spell/closed})
```

Since the `:conflicting` key is mentioned in the official document, it's nice to support this key officially. Therefore, @ikitommi mentioned about defining the key under `reitit.spec`.

Let me know if you have questions. I'm happy to provide more assistance on this one.

**Comment by wandersoncferreira** [comment]
Hello @dawran, I am actually very busy these days. If you can take it over, please go for it :)

Thanks for the detailed description of the problem.

**Comment by dawran6** [comment]
Welcome! I'll submit a PR in a moment.

**Comment by zelark** [comment]
Need to close it.

**Comment by ikitommi** [comment]
fixed in #444
